src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java

```diff
@@ -511,6 +512,14 @@ public ResultMessage execute(QueryState state, QueryOptions options, Dispatcher.
         {
             checkTrue(returningSelect.select.getLimit(options) == DataLimits.NO_LIMIT, NO_PARTITION_IN_CLAUSE_WITH_LIMIT, "SELECT", returningSelect.select.source);
         }
+        if (returningReferences != null)
+        {
+            for (RowDataReference dataRef : returningReferences)
+            {
+                TxnReference ref = dataRef.toTxnReference(options);
+                checkTrue(ref.kind == TxnReference.Kind.COLUMN, SELECT_WITH_REFERENCES_MUST_BE_A_COLUMN + ref.kind);
```

Comment:
nit: Worth making the error `public` and throwing a quick test in `TransactionStatementTest` to hit this?